test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def reset_config_loader_singleton():
+    ConfigLoader._instance = None
```

Comment:
probably we should write a reusable fixtures for ConfigLoader class and the rest we need here. I would rather use class method also instead of a new one written for test purposes. We have to also test reloading